metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/WorkflowExecutionFactory.java

```diff
@@ -91,6 +118,30 @@ private AbstractExecutablePlugin createWorkflowPlugin(
     return ExecutablePluginFactory.createPlugin(pluginMetadata);
   }
 
+  private boolean shouldRedirectsBePerformed(Dataset dataset,
+      ExecutablePluginType executablePluginType) {
+    boolean performRedirects = false;
+    try {
+      workflowUtils
+          .computePredecessorPlugin(executablePluginType, null, dataset.getDatasetId());
+    } catch (PluginExecutionNotAllowed pluginExecutionNotAllowed) {
+      //It means we need to do redirects because a new harvest was performed
+      performRedirects = true;
+      LOGGER.info("Predecessor plugin computation failed, so we want to create redirects",
+          pluginExecutionNotAllowed);
+    }
```

Comment:
Alright, perhaps I would just get the latests successful harvest and check the date with the latest index and compare.

Reply:
This should work, it would check if the latest harvest is the same with the linked to the requested plugin from db. So we are comparing if there is a new harvest executed therefore a redirect for an indexing operation would be applicable..

Reply:
After giving it some thought, I think I understand what you mean. But I do think that this is not the best way to test what you want to test. This method tests other things as well, and may have other criteria and is not meant for this purpose. Perhaps it is better to define exactly what it is we want to test and then test it. 

I guess that we want to test whether the previous harvest occurred after the previous index. You're getting the previous index below. So just get the previous harvest here and compare times?

Reply:
Okay I'm changing to the original suggestion.